components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1084,20 +869,349 @@ subroutine dropmixnuc( &
+               qcld_in = qcld_in + dact
+               nsource_out = nsource_out + dact*dtinv
+               raercol_cw_pt(mm) = raercol_cw_pt(mm) + dact  ! cloud-borne aerosol
+               raercol_pt(mm)    = raercol_pt(mm) - dact
+               dum = dumc*fm(imode)
+               do lspec = 1, nspec_amode(imode)
+                  mm = mam_idx(imode,lspec)
+                  spc_idx=lmassptr_amode(lspec,imode)
+                  dact    = dum*state_q_in(spc_idx) ! interstitial only
+                  raercol_cw_pt(mm) = raercol_cw_pt(mm) + dact  ! cloud-borne aerosol
+                  raercol_pt(mm)    = raercol_pt(mm) - dact
+               enddo
+            enddo
+         endif    !  cldn(icol,kk)-cldo(icol,kk) > 0.01_r8
+
+end subroutine update_from_newcld
+
+!===============================================================================
+
+subroutine update_from_cldn_profile(cldn_col_in,dtinv,wtke_col_in,zs,dz,temp_col_in, &
+           cs_col_in,csbot_cscen,state_q_col_in,raercol_nsav,raercol_cw_nsav,nsource_col, &
+           qcld,factnum_col,ekd,nact,mact)
+
+   ! input arguments
+   real(r8), intent(in) :: cldn_col_in(:)   ! cloud fraction [fraction]
+   real(r8), intent(in) :: dtinv      ! inverse time step for microphysics [s^{-1}]
+   real(r8), intent(in) :: wtke_col_in(:)   ! subgrid vertical velocity [m/s]
+   real(r8), intent(in) :: zs(:)            ! inverse of distance between levels [m^-1]
+   real(r8), intent(in) :: dz(:)         ! geometric thickness of layers [m]
+   real(r8), intent(in) :: temp_col_in(:)   ! temperature [K]
+   real(r8), intent(in) :: cs_col_in(:)     ! air density [kg/m^3]
+   real(r8), intent(in) :: csbot_cscen(:)   ! inverse normalized air density csbot(i)/cs(i,k) [dimensionless]
+   real(r8), intent(in) :: state_q_col_in(:,:)    ! aerosol mmrs [kg/kg]
+
+   real(r8), intent(inout) :: raercol_nsav(:,:)    ! single column of saved aerosol mass, number mixing ratios [#/kg or kg/kg]
+   real(r8), intent(inout) :: raercol_cw_nsav(:,:) ! same as raercol but for cloud-borne phase [#/kg or kg/kg]
+   real(r8), intent(inout) :: nsource_col(:)   ! droplet number mixing ratio source tendency [#/kg/s]
+   real(r8), intent(inout) :: qcld(:)  ! cloud droplet number mixing ratio [#/kg]
+   real(r8), intent(inout) :: factnum_col(:,:)  ! activation fraction for aerosol number [fraction]
+   real(r8), intent(inout) :: ekd(:)     ! diffusivity for droplets [m^2/s]
+   real(r8), intent(inout) :: nact(:,:)  ! fractional aero. number  activation rate [/s]
+   real(r8), intent(inout) :: mact(:,:)  ! fractional aero. mass    activation rate [/s]
+
+   ! local arguments
+
+   integer :: kk
+   integer :: kp1
+   integer  :: imode       ! mode counter variable
+   integer  :: lspec       ! species counter variable
+   integer  :: mm               ! local array index for MAM number, species
+   real(r8) :: dum, dumc        ! temporary variables
```

Comment:
It would be nice if we could get rid of `dum` variable names in favor of something more contextually relevant. :-)

Reply:
Yes, I had been intending on doing that for those variables in particular.  I've since gone ahead and done so.